src/ark/phenotyping/post_cluster_utils.py

```diff
@@ -167,3 +169,37 @@ def create_mantis_project(
         img_sub_folder="",
         seg_suffix_name=seg_suffix_name,
     )
+
+
+def generate_new_cluster_resolution(cell_table, cluster_col, new_cluster_col, cluster_mapping,
+                                    save_path):
+    """Add new column of more broad cell cluster assignments to the cell table.
+
+    Args:
+        cell_table (pd.DataFrame): cell table with clustered cell populations
+        cluster_col (str): column containing the cell phenotype
+        new_cluster_col (str): new column to create
+        cluster_mapping (dict): dictionary with keys detailing the new cluster names and values
+            explaining which cell types to group together
+        save_path (str): where to save the new cell table
+    """
+    # validation checks
+    misc_utils.verify_in_list(cluster_col=[cluster_col], cell_table_columns=cell_table.columns)
+    if new_cluster_col in cell_table.columns:
+        raise ValueError(f"The column {new_cluster_col} already exists in the cell table. "
+                         f"Please specify a different name for the new column.")
+
+    cluster_mapping_values = list(cluster_mapping.values())
+    cluster_list = list(itertools.chain.from_iterable(cluster_mapping_values))
```

Comment:
Should we add a check if a value in the dictionary is a single string not within a list? If someone accidentally just does a single string, it'll split the string into it's individual characters, for example `{"prev_cluster":"new_cluster"}` $\rightarrow$ `["n", "e", "w", "c", "l", "u", "s", "t", "e", "r"]`